crates/subspace-archiving/src/archiver.rs

```diff
@@ -185,6 +185,34 @@ pub struct NewArchivedSegment {
     pub object_mapping: Vec<PieceObjectMapping>,
 }
 
+impl NewArchivedSegment {
+    /// Returns all the object mappings in this archived segment as a lazy iterator.
+    pub fn global_object_mappings(&self) -> impl IntoIterator<Item = GlobalObject> + 'static {
```

Comment:
What is the point in returning `IntoIterator` instead of an iterator itself? It just forces you to call `into_iterator()` on every call:
```suggestion
    pub fn global_object_mappings(&self) -> impl Iterator<Item = GlobalObject> + 'static {
```

Reply:
I got the idiomatic Rust advice around the wrong way, arguments should be impl IntoIterator, return values should be impl Iterator: c491f13318